ssl/statem/statem_clnt.c

```diff
@@ -2423,9 +2423,13 @@ MSG_PROCESS_RETURN tls_process_key_exchange(SSL_CONNECTION *s, PACKET *pkt)
 MSG_PROCESS_RETURN tls_process_certificate_request(SSL_CONNECTION *s,
                                                    PACKET *pkt)
 {
-    /* Allocate and clear certificate validity flags */
-    OPENSSL_free(s->s3.tmp.valid_flags);
-    s->s3.tmp.valid_flags = OPENSSL_zalloc(s->ssl_pkey_num * sizeof(uint32_t));
+    /* Clear certificate validity flags */
+    if (s->s3.tmp.valid_flags)
```

Comment:
Make this `if (s->s3->tmp.valid_flags != NULL)`

Reply:
So done in https://github.com/openssl/openssl/pull/19312/commits/8be6985a9743a239684308b6258691d3489e52bf